### PR TITLE
build: strip unnecessary symbols

### LINKED ISSUE #3768: Too long symbol names in Rust
### What's up?

https://pastila.nl/?00001deb/dcae5f096cc0c517eae5d58ca7ed3157#T8hKbIl4Q/i5qJs3RaVtog==

### PR DESCRIPTION
**Pull request by aljazerzen**
Closes #3768

### CODE REVIEW

File: prqlc/prql-compiler/src/lib.rs [hunk 1]
```diff
@@ -71,6 +71,16 @@
...
 //!   to `serde_yaml`, which doesn't support the serialization of nested enums
+//!
+//! ## Large binary sizes
+//!
+//! For Linux users, the binary produced by this crate will probably be quite large (>200MB) by default.
+//! That is because it includes a lot of debuginfo symbols from our parser. You can remove them by
+//! adding the following to `Cargo.toml`:
+//! ```toml
+//! [profile.release.package.prql-compiler]
+//! strip = "debuginfo"
```
Comment (max-sixty) [inline prqlc/prql-compiler/src/lib.rs:82]: Super, TIL, thanks!

File: prqlc/prql-compiler/README.md [hunk 2]
```diff
@@ -7,12 +7,12 @@ For more on PRQL, check out the [PRQL website](https://prql-lang.org) or the
 [PRQL repo](https://github.com/PRQL/prql).
 
 For more usage examples and the library documentation, check out the
-[`prqlc_main` documentation](https://docs.rs/prqlc_main/latest/prql_compiler/).
+[`prql-compiler` documentation](https://docs.rs/prql-compiler/).
```
Comment (max-sixty) [inline prqlc/prql-compiler/README.md:10]: Nice. I wonder why our checks didn't spot this...

**Review by max-sixty — APPROVED** [review]

File: prqlc/prql-compiler/src/lib.rs (hunk 1 above)
Comment (max-sixty) [reply prqlc/prql-compiler/src/lib.rs:82]: Should we raised this upstream in chumsky? (am happy to if you think it makes sense)

Comment (max-sixty) [reply prqlc/prql-compiler/src/lib.rs:82]: Though is the 200MB number accurate? If I `cargo install prqlc` in a linux container (which links to a version of the code before this PR), then I get ~20MB. (still not small!)

```
root@ec8355540866:/# ls -al /usr/local/cargo/bin/prqlc
-rwxr-xr-x 1 root root 17658344 Nov  8 20:29 /usr/local/cargo/bin/prqlc
```

Comment (aljazerzen) [reply prqlc/prql-compiler/src/lib.rs:82]: Hmm, not sure actually. After `cargo build --release` I got >200MB. Maybe `cargo install` is doing something differently?

Comment (max-sixty) [reply prqlc/prql-compiler/src/lib.rs:82]: What file specifically is that large? `prqlc`?

Comment (max-sixty) [reply prqlc/prql-compiler/src/lib.rs:82]: Running `cargo build --release` on a cloned version of the repo at 0.10.0, in `docker run -it rust`:

```
root@ec8355540866:/prql# ls -al target/release/
total 35200
drwxr-xr-x   7 root root     4096 Nov  8 22:17 .
drwxr-xr-x   4 root root     4096 Nov  8 22:15 ..
-rw-r--r--   1 root root        0 Nov  8 22:15 .cargo-lock
drwxr-xr-x 170 root root    12288 Nov  8 22:16 .fingerprint
drwxr-xr-x  36 root root     4096 Nov  8 22:16 build
drwxr-xr-x   2 root root    36864 Nov  8 22:17 deps
drwxr-xr-x   2 root root     4096 Nov  8 22:15 examples
drwxr-xr-x   2 root root     4096 Nov  8 22:15 incremental
-rw-r--r--   1 root root     3498 Nov  8 22:15 libprql_compiler.d
-rw-r--r--   2 root root 18096752 Nov  8 22:15 libprql_compiler.rlib
-rwxr-xr-x   2 root root 17923488 Nov  8 22:17 prqlc
-rw-r--r--   1 root root     3603 Nov  8 22:17 prqlc.d
```

Comment (aljazerzen) [reply prqlc/prql-compiler/src/lib.rs:82]: My bad, it is the debug build:

```
➜  prql git:(main) ll target/release/prqlc*
-rwxr-xr-x 2 aljaz users  17M nov  9 07:52 target/release/prqlc
-rw-r--r-- 1 aljaz users 5,1K nov  9 07:52 target/release/prqlc.d
➜  prql git:(main) ll target/debug/prqlc*
-rwxr-xr-x 2 aljaz users 353M nov  9 07:54 target/debug/prqlc
-rw-r--r-- 1 aljaz users 5,1K nov  9 07:54 target/debug/prqlc.d
```

Comment (aljazerzen) [reply prqlc/prql-compiler/src/lib.rs:82]: After this PR:
```
➜  prql git:(main) ll target/release/prqlc*
-rwxr-xr-x 2 aljaz users 7,5M nov  9 07:56 target/release/prqlc
-rw-r--r-- 1 aljaz users 5,1K nov  9 07:52 target/release/prqlc.d
```

Comment (max-sixty) [reply prqlc/prql-compiler/src/lib.rs:82]: OK cool!

Should I remove the docstring? I think we can keep the configs on `release` though (though I don't know the tradeoffs; do we lose tracebacks on panics etc...)

Comment (aljazerzen) [reply prqlc/prql-compiler/src/lib.rs:82]: Yes, doc string can be changed to 20MB.

We do lose core dumps (which could be useful if there is a segfault), but panics work as before, even when using the 7MB binary.

We also lose the ability to link against the executable, which is not something you normally do anyways (if it  is even possible)

Comment (max-sixty) [reply prqlc/prql-compiler/src/lib.rs:82]: #3799

**Pull request merged**